### ISSUE TITLE
Delay applying the feed-sort-order setting until after feeds have been updated

### ISSUE THREAD

**Issue by graysky2** [newsboat/newsboat#1163]
Is it possible to have newsboat apply my `feed-sort-order lastupdated` setting after newboats finishes updating the feeds?  I am launching it via `newsboat -r` currently, and the effect is to apply the sort before all the feeds update.  Often, the results after the update are different.  Perhaps there is a way to do this now and I am just ignorant to it.

Note - I tried adding `auto-reload yes` to the config as well but that had no effect on delaying the sort.

**Comment by Minoru** [comment]
Do I understand correctly that the problem is: after reload, Newsboat re-sorts the feeds, which is inconvenient since the order changes suddenly?

**Comment by graysky2** [comment]
Yes, to me the expected behavior is for newboat to update all feeds first and then apply the value specified by `feed-sort-order` in the config.  This way, the user's options are applied to the most recent data rather than the old snapshot.

**Comment by Minoru** [comment]
Okay, so my understanding is *not* correct. Let me try again: the problem is that after reloading the feeds, Newsboat doesn't sort them again? If so, how do you verify that the sorting didn't happen?

**Comment by graysky2** [comment]
My config:
```
reload-threads 50
feed-sort-order lastupdated
```

When I run `newboat -r`, I see 0/0 for all my feeds because I deleted all read items prior.  Now, newsboat is updating the feeds, and let's say that several of them have new content.  They are not sorted to the top as I expect.  If I quit newsboat and reload it, I see the expected result, with the new content on the top of the list.

Did I explain it sufficiently well?

**Comment by Minoru** [comment]
Yes, thanks! So my second re-statement is correct, then.

I just checked with Newsboat 2.13, and it re-sorts the feedlist after reloading the feeds. A build from the current "master" segfaults, though :) I'll look into that as well. What version are *you* running, @graysky2?

**Comment by graysky2** [comment]
@Minoru - I am using 2.20.1 it seems:
```
% newsboat --version
newsboat 2.20.1 - https://newsboat.org/
Copyright (C) 2006-2015 Andreas Krennmair
Copyright (C) 2015-2020 Alexander Batischev
Copyright (C) 2006-2017 Newsbeuter contributors
Copyright (C) 2017-2020 Newsboat contributors

Newsboat is free software licensed under the MIT License. (Type `newsboat -vv' to see the full text.)
It bundles:
- JSON for Modern C++ library, licensed under the MIT License: https://github.com/nlohmann/json
- optional-lite library, licensed under the Boost Software License: https://github.com/martinmoene/optional-lite
- expected-lite library, licensed under the Boost Software License: https://github.com/martinmoene/expected-lite

newsboat 2.20.1
System: Linux 5.8.5-1-minimum (x86_64)
Compiler: g++ 10.1.0
ncurses: ncurses 6.2.20200212 (compiled with 6.2)
libcurl: libcurl/7.72.0 OpenSSL/1.1.1g zlib/1.2.11 zstd/1.4.5 libidn2/2.3.0 libpsl/0.21.1 (+libidn2/2.3.0) libssh2/1.9.0 nghttp2/1.41.0 (compiled with 7.71.1)
SQLite: 3.33.0 (compiled with 3.32.3)
libxml2: compiled with 2.9.10
```

**Comment by Minoru** [comment]
Got it, thanks!

I bisected the segfault to dd42cc37d21829df1d2322e2bfe7d7f5170e29ed. That's part of 2.20.1, so it's interesting that it's not segfaulting for you; perhaps it's dependent on the cache/number of urls/config/etc? I have to go sleep now, will turn that into a proper issue tomorrow.

**Comment by graysky2** [comment]
@Minoru - I built against master just now and my original issue is not present here.  The update happens first then the sort.  ge0e76

**Comment by Minoru** [comment]
Ah, it must've been #1089 then. Since the latest version works, I'm closing this issue. Thanks for submitting it!

(I'm still wrangling with the segfault. Can't reproduce it without my own cache.db. Will just submit a PR once I fixed it.)

**Comment by graysky2** [comment]
You're welcome, thanks for the time!
